src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/profile/Profile.java

```diff
@@ -0,0 +1,387 @@
+    this.teleSubjectiveValueWeightRange = null;
+  }
+
+  /**
+   * This function allows the user to clone an object Profile
+   *
+   * @return an object Profile
+   */
+  private Profile cloneProfile() {
+
+    Profile prof = new Profile();
+
+    prof.linearAvf = this.linearAvf;
+    prof.floorAreaWeightRange = this.floorAreaWeightRange;
+    prof.nbBedroomsWeightRange = this.nbBedroomsWeightRange;
+    prof.nbSleepingWeightRange = this.nbSleepingWeightRange;
+    prof.nbBathroomsWeightRange = this.nbBathroomsWeightRange;
+    prof.terraceWeightRange = this.terraceWeightRange;
+    prof.floorAreaTerraceWeightRange = this.floorAreaTerraceWeightRange;
+    prof.wifiSubjectiveValueWeightRange = this.wifiSubjectiveValueWeightRange;
+    prof.pricePerNightSubjectiveValueWeightRange = this.pricePerNightSubjectiveValueWeightRange;
+    prof.nbMinNightSubjectiveValueWeightRange = this.nbMinNightSubjectiveValueWeightRange;
+    prof.teleSubjectiveValueWeightRange = this.teleSubjectiveValueWeightRange;
+
+    return prof;
+  }
+
+  /**
+   * Gives the linearAVF of a Profile
+   *
+   * @return the LinearAVF of the Profile
+   */
+  public LinearAVF getLinearAVF() {
+    return this.linearAvf;
+  }
+
+  /**
+   * Gives the subjective value weight of a criterion
+   *
+   * @param crit the criterion we want to know the value
+   * @return the subjective value weight
+   */
+  public Range<Double> getWeightRange(Criterion crit) {
+    switch (crit) {
+      case TELE:
+        return teleSubjectiveValueWeightRange;
+      case TERRACE:
+        return terraceWeightRange;
+      case WIFI:
+        return wifiSubjectiveValueWeightRange;
+      case FLOOR_AREA:
+        return floorAreaWeightRange;
+      case FLOOR_AREA_TERRACE:
+        return floorAreaTerraceWeightRange;
+      case NB_BATHROOMS:
+        return nbBathroomsWeightRange;
+      case NB_BEDROOMS:
+        return nbBedroomsWeightRange;
+      case NB_SLEEPING:
+        return nbSleepingWeightRange;
+      case NB_MIN_NIGHT:
+        return nbMinNightSubjectiveValueWeightRange;
+      case PRICE_PER_NIGHT:
+        return pricePerNightSubjectiveValueWeightRange;
+      default:
+        throw new IllegalArgumentException();
+    }
+  }
+
+  /**
+   * Get the middle of the given Range
+   *
+   * @param range the Range
+   * @return the middle of the given Range of Doubles
+   */
+  static double getMiddleOfRange(Range<Double> range) {
+    checkNotNull(range);
+    return (range.upperEndpoint() + range.lowerEndpoint()) / 2;
+  }
+
+  /**
+   * Get the middle of the range of the given criterion
+   *
+   * @param crit the Criterion
+   * @return the middle of the Range of the given Criterion
+   */
+  double getMiddleOfRange(Criterion crit) {
+    return getMiddleOfRange(this.getWeightRange(crit));
+  }
+
+  /**
+   * Sets the linearAVF of a Profile
+   *
+   * @return Profile with its LinearAVF set
+   */
+  public Profile setLinearAVF(LinearAVF newLinearAvf) {
```

Comment:
Voir ci-dessus concernant ce nom.

Reply:
fixed in e2f1008d8408d18722b97fa759860bade4406f74

Reply:
fixed in bda21cc093cc6b4bfb222fa95e12206379ea89a0

Reply:
Fixed in bda21cc093cc6b4bfb222fa95e12206379ea89a0